Index: votingServer/src/main/java/voting/controllers/SphereController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- votingServer/src/main/java/voting/controllers/SphereController.java	(revision 7430ffafbf7849e9a44f39cac84258257183346a)
+++ votingServer/src/main/java/voting/controllers/SphereController.java	(revision )
@@ -3,6 +3,7 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.MediaType;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.ResponseBody;
@@ -24,5 +25,11 @@
     @ResponseBody
     public List<Sphere> getAllActs(){
         return sphereDao.getAll();
+    }
+
+    @RequestMapping(value="/spheres/{name}")
+    @ResponseBody
+    public Sphere getSphereByName(@PathVariable String name) {
+        return sphereDao.getByName(name);
     }
 }
Index: votingServer/src/main/java/voting/models/SphereDao.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- votingServer/src/main/java/voting/models/SphereDao.java	(revision 7430ffafbf7849e9a44f39cac84258257183346a)
+++ votingServer/src/main/java/voting/models/SphereDao.java	(revision )
@@ -22,4 +22,11 @@
     public List<Sphere> getAll() {
         return entityManager.createQuery("from Sphere").getResultList();
     }
+
+    public Sphere getByName(String name){
+        return (Sphere) entityManager.createQuery(
+                "from Sphere where name = :name")
+                .setParameter("name", name)
+                .getSingleResult();
+    }
 }
Index: votingServer/src/main/java/voting/controllers/ActController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- votingServer/src/main/java/voting/controllers/ActController.java	(revision 7430ffafbf7849e9a44f39cac84258257183346a)
+++ votingServer/src/main/java/voting/controllers/ActController.java	(revision )
@@ -29,7 +29,7 @@
         return actDao.getAll();
     }
 
-    @RequestMapping(value="/get-act-by-sphere/{id}")
+    @RequestMapping(value="/acts/{id}")
     @ResponseBody
     public List<Act> getActBySphere(@PathVariable int id) {
         List<Act> selectedActs = new ArrayList<>();
